src/views/connectionUI.ts

```diff
@@ -269,6 +269,31 @@ export class ConnectionUI {
         });
     }
 
+    public createProfileWithDifferentCredentials(connection: IConnectionCredentials): Promise<IConnectionCredentials> {
+
+        return new Promise<IConnectionCredentials>((resolve, reject) => {
+            this.promptForRetryConnectWithDifferentCredentials().then(result => {
+                if (result) {
+                    let connectionWithoutCredentials = Object.assign({}, connection, { user: '', password: '', emptyPasswordInput: false });
+                    ConnectionCredentials.ensureRequiredPropertiesSet(
+                        connectionWithoutCredentials,
+                        true,
```

Comment:
Minor: can you add comments for what these values are? A bunch of true / false / true makes it hard to read.